code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,378 @@
+		return FALSE
+	if(!user.drop_item())
+		to_chat(user, "<span class='warning'>[item_to_insert] is stuck to your hand!</span>")
+		return FALSE
+	item_to_insert.forceMove(src)
+	RegisterSignal(item_to_insert, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	LAZYADD(inserted_items, item_to_insert)
+	calculate_tub_capacity()
+	update_washing_state()
+	to_chat(user, "<span class='notice'>You insert [item_to_insert] into [src].</span>")
+	return TRUE
 
-		if(length(contents) < 5)
-			if(state in list(OPEN_EMPTY, OPEN_FULL))
-				user.drop_item()
-				W.forceMove(src)
-				state = OPEN_FULL
-			else
-				to_chat(user, "<span class='warning'>The door is closed!</span>")
-		else
-			to_chat(user, "<span class='warning'>[src] is full!</span>")
-		update_icon(UPDATE_ICON_STATE)
+/*
+	* # check_tub_contents()
+	*
+	* Managing tons of contents inside the object is messy business, you never know when something is deleted, teleported, or moved into your obj
+	* without your machine knowing, this clears bad references for GC/tracking purposes and adds references when something is added randomly to
+	* our contents. This ensures we don't nullspace shit (such as a mob dropping a wrench or something) and spit it all back out afterwards even
+	* if we didn't add it our selves!
+	*
+	* This proc is called via COMSIG_PARENT_QDELETING for objects we put inside through the insert proc
+*/
+/obj/machinery/washing_machine/proc/check_tub_contents()
+	if(LAZYLEN(inserted_items))
+		for(var/obj/item/I in inserted_items)
+			if(QDELETED(I) || I.loc != src)
+				if(!QDELETED(I))
+					UnregisterSignal(I, COMSIG_PARENT_QDELETING)
+				LAZYREMOVE(inserted_items, I)
+	if(LAZYLEN(inserted_mobs))
+		for(var/mob/living/L in inserted_mobs)
+			if(QDELETED(L) || L.loc != src)
+				if(!QDELETED(L))
+					UnregisterSignal(L, COMSIG_PARENT_QDELETING)
+				LAZYREMOVE(inserted_mobs, L)
+	for(var/obj/item/I in contents)
+		if(!LAZYIN(inserted_items, I))
+			LAZYADD(inserted_items, I)
+			RegisterSignal(I, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	for(var/mob/living/L in contents)
+		if(!LAZYIN(inserted_mobs, L))
+			LAZYADD(inserted_mobs, L)
+			RegisterSignal(L, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	calculate_tub_capacity()
+	update_washing_state()
+
+/// Moves all Items/Mobs out of the laundry machine onto the Washing Machine's Loc, unregisters signals, then nulls the tracking lists
+/obj/machinery/washing_machine/proc/eject_tub_contents()
+	if(bloody_mess)
+		playsound(loc, 'sound/effects/splat.ogg', 50, TRUE)
+	check_tub_contents() // one last time, we wanna make sure nothing got teleported out
+	if(LAZYLEN(inserted_items))
+		for(var/obj/item/I in inserted_items)
+			if(bloody_mess)
+				I.add_blood() // what a mess!
+			I.forceMove(loc)
+			UnregisterSignal(I, COMSIG_PARENT_QDELETING)
```

Comment:
went ahead and added these signals to tracking on inserted atom moveables. This will eliminate two iterations of the contents of the washing machine every cycle (unless stuff gets teleported out mid-cycle).